core/src/avm2/globals/flash/geom/matrix.rs

```diff
@@ -0,0 +1,180 @@
+    activation: &mut Activation<'_, 'gc, '_>,
+    a: f64,
+    b: f64,
+    c: f64,
+    d: f64,
+    tx: f64,
+    ty: f64,
+) -> Result<(), Error> {
+    this.set_property(
+        &QName::new(Namespace::public(), "a").into(),
+        a.into(),
+        activation,
+    )?;
+    this.set_property(
+        &QName::new(Namespace::public(), "b").into(),
+        b.into(),
+        activation,
+    )?;
+    this.set_property(
+        &QName::new(Namespace::public(), "c").into(),
+        c.into(),
+        activation,
+    )?;
+    this.set_property(
+        &QName::new(Namespace::public(), "d").into(),
+        d.into(),
+        activation,
+    )?;
+    this.set_property(
+        &QName::new(Namespace::public(), "tx").into(),
+        tx.into(),
+        activation,
+    )?;
+    this.set_property(
+        &QName::new(Namespace::public(), "ty").into(),
+        ty.into(),
+        activation,
+    )?;
+
+    Ok(())
+}
+
+/// Implements `flash.geom.Matrix`'s instance constructor.
+pub fn instance_init<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(mut this) = this {
+        let a = args
+            .get(0)
+            .unwrap_or(&1f64.into())
+            .coerce_to_number(activation)?;
+
+        let b = args
+            .get(1)
+            .unwrap_or(&0f64.into())
+            .coerce_to_number(activation)?;
+
+        let c = args
+            .get(2)
+            .unwrap_or(&0f64.into())
+            .coerce_to_number(activation)?;
+
+        let d = args
+            .get(3)
+            .unwrap_or(&1f64.into())
+            .coerce_to_number(activation)?;
+
+        let tx = args
+            .get(4)
+            .unwrap_or(&0f64.into())
+            .coerce_to_number(activation)?;
+
+        let ty = args
+            .get(5)
+            .unwrap_or(&0f64.into())
+            .coerce_to_number(activation)?;
+
+        set_values(&mut this, activation, a, b, c, d, tx, ty)?;
+    }
+    Ok(Value::Undefined)
+}
+
+/// Implements `flash.geom.Matrix`'s class initializer.
+pub fn class_init<'gc>(
+    _activation: &mut Activation<'_, 'gc, '_>,
+    _this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    Ok(Value::Undefined)
+}
+
+/// Implements `identity'
+pub fn identity<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(mut this) = this {
+        set_values(&mut this, activation, 1.0, 0.0, 0.0, 1.0, 0.0, 0.0)?;
+    }
+    Ok(Value::Undefined)
+}
+
+/// Implements `scale'
+pub fn scale<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(mut this) = this {
+        let sx = args
+            .get(0)
+            .expect("Called with wrong number of args")
```

Comment:
In other classes (e.g. `Point`), it looks like we handle missing arguments by either returning early, or coming up with a default value. However, my understanding is that this case should never be hit (since the argument is required) unless we're executing a malformed SWF.

Reply:
When I tried to call `.scale()` with no arguments the following exception occurred:
```
ArgumentError: Error #1063: Argument count mismatch on flash.geom::Matrix/scale(). Expected 2, got 0.
```
Maybe we should mimic this by returning an `Err` instead of panicking.

Also, can you please prefix your commit message with `avm2:` as described in https://github.com/ruffle-rs/ruffle/blob/master/CONTRIBUTING.md#commit-message-guidelines?

Reply:
(In general (long-term), pretty sure we should do this either as part of `fn exec()` infrastructure, with attaching typing metadata to native methods - or some macro-based solution that removes the boilerplate and redundancies automatically, like in RustPython.)
But yeah, for now, let's not panic :)